### PR TITLE
chore(CODEOWNERS): Remove unused teams and add cloud-samples-infra

### PR DESCRIPTION
**Pull request by grayside**
## Description

Fixes b/368475360 and cleans up unused teams/redundant entries.

As part of this, there's a new row for `/*`, this is a way to give CODEOWNERS authority to root files without the entire repo: the cloud-samples-infra team is not scoped for reviewing terraform sample changes.

GitHub is now marking this change as invalid because the new team needs to be added as a repo admin. Requesting a reviewer from terraform-samples-git-admins take care of that.

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [ ] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [ ] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [ ] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [ ] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [ ] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [ ] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Review by iennae — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,28 +1,24 @@
...
 /cloudvpn/              @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
 /composer/              @terraform-google-modules/cloud-dpes-composer @terraform-google-modules/terraform-samples-reviewers
 /compute/               @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
 /dns/                   @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
-/eventarc/              @terraform-google-modules/torus-dpe @terraform-google-modules/terraform-samples-reviewers
-/functions/             @terraform-google-modules/torus-dpe @terraform-google-modules/terraform-samples-reviewers
 /gke/                   @terraform-google-modules/eks-team @terraform-google-modules/terraform-samples-reviewers
 /lb/                    @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
-/looker/		@terraform-google-modules/cloud-looker-docs @terraform-google-modules/terraform-samples-reviewers
+/looker/		            @terraform-google-modules/cloud-looker-docs @terraform-google-modules/terraform-samples-reviewers
```
Comment (glasnt) [inline .github/CODEOWNERS:17]:
```suggestion
/looker/		@terraform-google-modules/cloud-looker-docs @terraform-google-modules/terraform-samples-reviewers
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,28 +1,24 @@
...
 /media_cdn/             @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
 /network_connectivity/  @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
 /privateca/             @terraform-google-modules/dee-infra @terraform-google-modules/terraform-samples-reviewers
-/run/                   @terraform-google-modules/torus-dpe @terraform-google-modules/terraform-samples-reviewers
```
Comment (glasnt) [inline .github/CODEOWNERS:22]: Why are these being removed?

**Review by glasnt — APPROVED** [review]
added cloud-samples-infra to write permissions, some pending comments

File: .github/CODEOWNERS (hunk 2 above)
Comment (grayside) [reply .github/CODEOWNERS:22]: torus-dpe team is no longer in use and being removed from repositories. Once torus-dpe is removed from this row, the value of keeping this rule as an override seems missing. Is there a need to keep something in place?

**Comment by msampathkumar** [comment]
Assigning to @glasnt, to check and merge this PR.

File: .github/CODEOWNERS (hunk 2 above)
Comment (glasnt) [reply .github/CODEOWNERS:22]: I see; I believe the intent here is that samples-reviews from the top level will catch this. 

I think this *isn't* entirely best practice, and not something we should be suggesting others do, but given this isn't one of our Big Four, I think it's fine? 

(Strictly speaking samples should never be added without expert reviewers, but the fact that in this specific case the expert reviewers moved teams to the superteam, I think it's fine)

Comment (grayside) [reply .github/CODEOWNERS:22]: Clarifying, the big four have this, and the intent is that cloud-samples-infra is able to approve things like changes to README, Makefile, and CONTRIBUTING without having authority over the samples themselves. This PR should only have removed unused CODEOWNERS of samples, not moved authority around.

**Pull request merged**